Enhancements to Security docs Basic authentication tutorial (#30361)
quarkusio/quarkus: docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc

== What you will build

The steps in this tutorial guide you through building an application that provides the following endpoints to demonstrate different authorization policies:

[cols="20%,40% ",options="header"]
|===
|Endpoint | Description 
|`/api/public`| The `/api/public` endpoint can be accessed anonymously.
|`/api/admin`| The `/api/admin` endpoint is protected with role-based access control (RBAC), and only users who have been granted the `admin` role can access it.
At this endpoint, the `@RolesAllowed` annotation is used to enforce the access constraint declaratively.

> |`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it.

Reviewer: You're missing the sentence about what it returns (it should be just a string with the username - not a JSON document)

Author: Yeah, I just add it. THX :)

== What you will build

The steps in this tutorial guide you through building an application that provides the following endpoints to demonstrate different authorization policies:

[cols="20%,40% ",options="header"]
|===
|Endpoint | Description 
|`/api/public`| The `/api/public` endpoint can be accessed anonymously.
|`/api/admin`| The `/api/admin` endpoint is protected with role-based access control (RBAC), and only users who have been granted the `admin` role can access it.

> At this endpoint, the `@RolesAllowed` annotation is used to enforce the access constraint declaratively.

Reviewer 2: Maybe 
```suggestion
At this endpoint, the `@RolesAllowed` annotation enforces the access constraint declaratively.
```

> |`/api/users/me`| The `/api/users/me` endpoint is protected with RBAC and only users that have been granted the `user` role can access it.

Reviewer 2: Maybe 
```suggestion
|`/api/users/me`| The `/api/users/me` endpoint is protected by RBAC. Only users that have the `user` role can access the endpoint.
```

> This endpoint returns a string value containing the caller's username.

Reviewer 2: Maybe 
```suggestion
This endpoint returns the caller's username as a string.
```

The `security-jpa` extension initializes only if there is a single entity annotated with `@UserDefinition`.

<1> The `@UserDefinition` annotation must be present on a single entity and can be either a regular Hibernate ORM entity or a Hibernate ORM with Panache entity.
<2> Indicates the field used for the username.

> <3> Indicates the field used for the password, which defaults to using bcrypt hashed passwords, but you can also configure it for plain text or custom passwords.

Reviewer 2: ```suggestion
<3> Indicates the field used for the password. By default, it uses bcrypt-hashed passwords. You can configure it to use plain text or custom passwords.
```

By default, passwords are stored and hashed by using https://en.wikipedia.org/wiki/Bcrypt[bcrypt] under the 
https://en.wikipedia.org/wiki/Crypt_(C)[Modular Crypt Format] (MCF).

> When you need to create a hashed password, provide the convenient `String BcryptUtil.bcryptHash(String password)`

Reviewer: No, the "we" means Quarkus - the user doesn't have to provide it. 
The original meaning was correct, maybe to improve I would change "we provide" to "you may use"

Author: OK, applying
